diff --git a/lib/kobject_uevent.c b/lib/kobject_uevent.c
index da602353..a42a358f 100644
--- a/lib/kobject_uevent.c
+++ b/lib/kobject_uevent.c
@@ -176,6 +176,18 @@ static int kobject_action_args(const char *buf, size_t count,
 	return r;
 }
 
+u64 uevent_next_seqnum(void)
+{
+	u64 seq;
+
+	mutex_lock(&uevent_sock_mutex);
+	seq = ++uevent_seqnum;
+	mutex_unlock(&uevent_sock_mutex);
+
+	return seq;
+}
+EXPORT_SYMBOL_GPL(uevent_next_seqnum);
+
 /**
  * kobject_synth_uevent - send synthetic uevent with arguments
  *
