diff --git a/arch/mips/kernel/process.c b/arch/mips/kernel/process.c
index ea563bfe..4abdb6eb 100644
--- a/arch/mips/kernel/process.c
+++ b/arch/mips/kernel/process.c
@@ -360,6 +360,8 @@ static inline int is_sp_move_ins(union mips_instruction *ip, int *frame_size)
 
 	if (ip->i_format.opcode == addiu_op ||
 	    ip->i_format.opcode == daddiu_op) {
+		if (ip->i_format.simmediate > 0)
+			return 0;
 		*frame_size = -ip->i_format.simmediate;
 		return 1;
 	}
