diff --git a/mm/page_alloc.c b/mm/page_alloc.c
index 7163cf68..7a7d668c 100644
--- a/mm/page_alloc.c
+++ b/mm/page_alloc.c
@@ -6213,7 +6213,7 @@ static void __ref alloc_node_mem_map(struct pglist_data *pgdat)
 		mem_map = NODE_DATA(0)->node_mem_map;
 #if defined(CONFIG_HAVE_MEMBLOCK_NODE_MAP) || defined(CONFIG_FLATMEM)
 		if (page_to_pfn(mem_map) != pgdat->node_start_pfn)
-			mem_map -= offset;
+			mem_map -= offset + (pgdat->node_start_pfn - ARCH_PFN_OFFSET);
 #endif /* CONFIG_HAVE_MEMBLOCK_NODE_MAP */
 	}
 #endif
