diff --git a/drivers/spi/spi.c b/drivers/spi/spi.c
index 1031c8e3..bfcf28fd 100644
--- a/drivers/spi/spi.c
+++ b/drivers/spi/spi.c
@@ -741,7 +741,7 @@ static void spi_set_cs(struct spi_device *spi, bool enable)
 		enable = !enable;
 
 	if (gpio_is_valid(spi->cs_gpio)) {
-		gpio_set_value(spi->cs_gpio, !enable);
+		gpio_set_value_cansleep(spi->cs_gpio, !enable);
 		/* Some SPI masters need both GPIO CS & slave_select */
 		if ((spi->controller->flags & SPI_MASTER_GPIO_SS) &&
 		    spi->controller->set_cs)
