diff --git a/drivers/platform/Kconfig b/drivers/platform/Kconfig
index c11db8bc..0283f0b4 100644
--- a/drivers/platform/Kconfig
+++ b/drivers/platform/Kconfig
@@ -8,3 +8,5 @@ endif
 source "drivers/platform/goldfish/Kconfig"
 
 source "drivers/platform/chrome/Kconfig"
+
+source "drivers/platform/mikrotik/Kconfig"
diff --git a/drivers/platform/Makefile b/drivers/platform/Makefile
index d3a66302..ad290c30 100644
--- a/drivers/platform/Makefile
+++ b/drivers/platform/Makefile
@@ -8,3 +8,4 @@ obj-$(CONFIG_MIPS)		+= mips/
 obj-$(CONFIG_OLPC)		+= olpc/
 obj-$(CONFIG_GOLDFISH)		+= goldfish/
 obj-$(CONFIG_CHROME_PLATFORMS)	+= chrome/
+obj-$(CONFIG_MIKROTIK)		+= mikrotik/
