diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index 59f55427..5682d792 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -1073,9 +1073,6 @@ config FW_ARC
 config ARCH_MAY_HAVE_PC_FDC
 	bool
 
-config BOOT_RAW
-	bool
-
 config CEVT_BCM1480
 	bool
 
@@ -2974,6 +2971,18 @@ choice
 		bool "Extend builtin kernel arguments with bootloader arguments"
 endchoice
 
+config BOOT_RAW
+	bool "Enable the kernel to be executed from the load address"
+	default n
+	help
+	 Allow the kernel to be executed from the load address for
+	 bootloaders which cannot read the ELF format. This places
+	 a jump to start_kernel at the load address.
+
+	 If unsure, say N.
+
+
+
 endmenu
 
 config LOCKDEP_SUPPORT
