diff --git a/drivers/mtd/redboot.c b/drivers/mtd/redboot.c
index 7623ac5f..663c7aeb 100644
--- a/drivers/mtd/redboot.c
+++ b/drivers/mtd/redboot.c
@@ -289,9 +289,16 @@ static int parse_redboot_partitions(struct mtd_info *master,
 	return ret;
 }
 
+static const struct of_device_id redboot_parser_of_match_table[] = {
+	{ .compatible = "ecoscentric,redboot-fis-partitions" },
+	{},
+};
+MODULE_DEVICE_TABLE(of, redboot_parser_of_match_table);
+
 static struct mtd_part_parser redboot_parser = {
 	.parse_fn = parse_redboot_partitions,
 	.name = "RedBoot",
+	.of_match_table = redboot_parser_of_match_table,
 };
 module_mtd_part_parser(redboot_parser);
 
