diff --git a/include/net/netfilter/nf_flow_table.h b/include/net/netfilter/nf_flow_table.h
index baa64d3a..3a461a5a 100644
--- a/include/net/netfilter/nf_flow_table.h
+++ b/include/net/netfilter/nf_flow_table.h
@@ -76,6 +76,7 @@ struct flow_offload_tuple_rhash {
 #define FLOW_OFFLOAD_DYING	0x4
 #define FLOW_OFFLOAD_TEARDOWN	0x8
 #define FLOW_OFFLOAD_HW		0x10
+#define FLOW_OFFLOAD_KEEP	0x20
 
 struct flow_offload {
 	struct flow_offload_tuple_rhash		tuplehash[FLOW_OFFLOAD_DIR_MAX];
diff --git a/net/netfilter/nf_flow_table_core.c b/net/netfilter/nf_flow_table_core.c
index 421e0aa4..9a7ea794 100644
--- a/net/netfilter/nf_flow_table_core.c
+++ b/net/netfilter/nf_flow_table_core.c
@@ -359,7 +359,7 @@ static int nf_flow_offload_gc_step(struct nf_flowtable *flow_table)
 		if (!teardown)
 			nf_ct_offload_timeout(flow);
 
-		if (nf_flow_in_hw(flow) && !teardown)
+		if ((flow->flags & FLOW_OFFLOAD_KEEP) && !teardown)
 			continue;
 
 		if (nf_flow_has_expired(flow) || teardown)
