diff --git a/include/net/netfilter/nf_flow_table.h b/include/net/netfilter/nf_flow_table.h
index 3a461a5a..7cfc3147 100644
--- a/include/net/netfilter/nf_flow_table.h
+++ b/include/net/netfilter/nf_flow_table.h
@@ -81,9 +81,10 @@ struct flow_offload_tuple_rhash {
 struct flow_offload {
 	struct flow_offload_tuple_rhash		tuplehash[FLOW_OFFLOAD_DIR_MAX];
 	u32					flags;
+	u32					timeout;
 	union {
 		/* Your private driver data here. */
-		u32		timeout;
+		void *priv;
 	};
 };
 
