diff --git a/drivers/net/phy/phy_device.c b/drivers/net/phy/phy_device.c
index 5d557a00..142952ba 100644
--- a/drivers/net/phy/phy_device.c
+++ b/drivers/net/phy/phy_device.c
@@ -1111,6 +1111,9 @@ void phy_detach(struct phy_device *phydev)
 	struct module *ndev_owner = dev->dev.parent->driver->owner;
 	struct mii_bus *bus;
 
+	if (phydev->drv && phydev->drv->detach)
+		phydev->drv->detach(phydev);
+
 	if (phydev->sysfs_links) {
 		sysfs_remove_link(&dev->dev.kobj, "phydev");
 		sysfs_remove_link(&phydev->mdio.dev.kobj, "attached_dev");
diff --git a/include/linux/phy.h b/include/linux/phy.h
index 8b685070..87e70092 100644
--- a/include/linux/phy.h
+++ b/include/linux/phy.h
@@ -561,6 +561,12 @@ struct phy_driver {
 	 */
 	int (*did_interrupt)(struct phy_device *phydev);
 
+	/*
+	 * Called before an ethernet device is detached
+	 * from the PHY.
+	 */
+	void (*detach)(struct phy_device *phydev);
+
 	/* Clears up any memory if needed */
 	void (*remove)(struct phy_device *phydev);
 
