diff --git a/drivers/tty/serial/serial_core.c b/drivers/tty/serial/serial_core.c
index 230e515b..5c2e4acf 100644
--- a/drivers/tty/serial/serial_core.c
+++ b/drivers/tty/serial/serial_core.c
@@ -435,6 +435,9 @@ uart_get_baud_rate(struct uart_port *port, struct ktermios *termios,
 		break;
 	}
 
+	if (tty_termios_baud_rate(termios) == 2500000)
+		return 250000;
+
 	for (try = 0; try < 2; try++) {
 		baud = tty_termios_baud_rate(termios);
 
